daemon/api_system_volumes.go

```diff
@@ -87,3 +98,29 @@ func postSystemVolumesActionGenerateRecoveryKey(c *Command) Response {
 		"key-id":       keyID,
 	})
 }
+
+var keyIDRequiredErr = BadRequest("system volume action requires key-id to be provided")
+
+func postSystemVolumesActionReplaceRecoveryKey(c *Command, req *systemVolumesActionRequest) Response {
+	if req.KeyID == "" {
+		return keyIDRequiredErr
+	}
+
+	st := c.d.overlord.State()
+	st.Lock()
+	defer st.Unlock()
+
+	if len(req.Keyslots) == 0 {
+		// target default-recovery key slots by default if no key slot targets are specified
+		req.Keyslots = append(req.Keyslots,
+			fdestate.KeyslotTarget{ContainerRole: "system-data", Name: "default-recovery"},
+			fdestate.KeyslotTarget{ContainerRole: "system-save", Name: "default-recovery"},
+		)
+	}
+
+	chg, err := fdestateReplaceRecoveryKey(st, req.KeyID, req.Keyslots)
+	if err != nil {
+		return BadRequest("cannot change recovery key: %v", err)
+	}
+	return AsyncResponse(nil, chg.ID())
```

Comment:
you'll prob want to call `st.EnsureBefore(0)`, otherwise the change won't be picked up immediately.